Plots/Bar/NCL_unique_5.py

```diff
@@ -0,0 +1,91 @@
+"""
+NCL_unique_5.py
+===============
+Concepts illustrated:
+  - Drawing multiple bar charts on a page
+  - Drawing three custom legends outside a bar chart
+  - Using bar charts to draw standard deviations for four time series
+  - Drawing a time series plot
+
+See following URLs to see the reproduced NCL plot & script:
+    - Original NCL script: https://www.ncl.ucar.edu/Applications/Scripts/unique_5.ncl
+    - Original NCL plot: https://www.ncl.ucar.edu/Applications/Images/unique_5_lg.png
+
+"""
+
+###############################################################################
+# Import packages:
+import numpy as np
+import matplotlib.pyplot as plt
+import random
+
+from geocat.viz import util as gvutil
+
+###############################################################################
+# Generate labels:
+x = np.array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12])
+labels = ['Jan', 'Feb', 'Mar', 'Apr', 'May',
+          'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec']
+
+# Generate random data:
+obs = []
+ccsm2_t42 = []
+ccsm3_t42 = []
+ccsm3_t85 = []
+
+for i in range(12):
+    obs.append(random.uniform(0.4, 1.2))
+    ccsm2_t42.append(random.uniform(0.4, 1.2))
+    ccsm3_t42.append(random.uniform(0.4, 1.2))
+    ccsm3_t85.append(random.uniform(0.4, 1.2))
+
+###############################################################################
+# Create the custom color list.
+color_list = ['red', 'lightsteelblue', 'blue', 'lime']
+
+###############################################################################
+# Specify some plot settings to use in the plot:
+
+# Title settings
+title = 'Nino3.4 Monthly Standard Deviation'
+title_fontsize = 16
+
+# Axis Settings
+plot_y_max = 1.2
+
+###############################################################################
+# Plot bar chart with a legend
+
+# Generate figure
+plt.figure(2, figsize=(6, 5))
+ax = plt.gca()
+
+# Set width of each column
+w = 0.15
+
+# Create subplots for each category
+sub = plt.subplot(111)
+sub.bar(x+w, obs, width=0.15, color=color_list[0], edgecolor='black', linewidth=0.25, align='center')
+sub.bar((x+(2*w)), ccsm2_t42, width=0.15, color=color_list[1], edgecolor='black', linewidth=0.25, align='center')
+sub.bar(x+3*w, ccsm3_t42, width=0.15, color=color_list[2], edgecolor='black', linewidth=0.25, align='center')
+sub.bar(x+4*w, ccsm3_t85, width=0.15, color=color_list[3], edgecolor='black', linewidth=0.25, align='center')
+
+# Add label to each bar
```

Comment:
This `for-loop` seems to work well, but I would personally not use it. Instead I would remove it and modify **Line 81** and **Line 84** as follows to plot the same month names on **x-axis**:

`gvutil.set_axes_limits_and_ticks(ax, ylim=(0.4, plot_y_max), xticks=x, xticklabels=labels, yticks=np.linspace(0.4, plot_y_max, 5))`

`gvutil.add_major_minor_ticks(ax, x_minor_per_major=1, y_minor_per_major=4, labelsize=12)`

This way, it would also solve another issue: The current version of this plot does not have major ticks for months on top and bottom x-axis but **NCL**'s original plot has. The above two lines add those major ticks for months into the plot. However, there is also an issue with my solution here: Major ticks are displayed in the beginning of every 4-bar group, but NCL displays those major ticks in the middle of each 4-bar group. You may further play with it to walk around this issue.

Reply:
It looks like the NCL graph displays them at the beginning as well, the only difference I can spot with the python version is that the last tick on the top and bottom axes is small. Do you know why this is?
![Inkednew graph_LI](https://user-images.githubusercontent.com/36773444/83056816-e1679300-a00a-11ea-9824-f77111828d7f.jpg)

Reply:
1. "the only difference I can spot with the python version is that the last tick on the top and bottom axes is small. Do you know why this is?"

You can play with the `xlim` argument of the following function to make disappear that final minor tick:

`gvutil.set_axes_limits_and_ticks(ax, ylim=(0.4, plot_y_max), xticks=x, xticklabels=labels, yticks=np.linspace(0.4, plot_y_max, 5))`

2. "It looks like the NCL graph displays them at the beginning as well"

The [NCL plot](https://www.ncl.ucar.edu/Applications/Images/unique_5_lg.png) given in the corresponding issue seems to be placing those tick in the middle of each 4-bar group (i.e. centers the bar groups to each month tick), Please see:

![image](https://user-images.githubusercontent.com/32553057/83061201-d9fbb600-a019-11ea-8880-dcb2e02212ae.png)

However, I am not sure if we should worry about it. Hi @jukent do you have a strong feeling about it?

Reply:
I like the way they look in the middle better.

Here is an example of how to align them in the center:
https://stackoverflow.com/questions/31067389/matplotlib-barplot-generic-way-of-setting-the-xticklabels-in-the-middle-of-the